tutorials/tree/copytree.C

```diff
@@ -10,31 +10,31 @@
 ///
 /// \author Rene Brun
 
+// Load the library at macro parsing time: we need this to use its content in the code
 R__LOAD_LIBRARY($ROOTSYS/test/libEvent.so)
 
-void copytree() {
-   //Get old file, old tree and set top branch address
-   TFile *oldfile;
+void copytree()
+{
+
    TString dir = "$ROOTSYS/test/Event.root";
    gSystem->ExpandPathName(dir);
-   if (!gSystem->AccessPathName(dir))
-       {oldfile = new TFile("$ROOTSYS/test/Event.root");}
-   else {oldfile = new TFile("./Event.root");}
-   TTree *oldtree = (TTree*)oldfile->Get("T");
-   Event *event   = new Event();
-   oldtree->SetBranchAddress("event",&event);
-   oldtree->SetBranchStatus("*",0);
-   oldtree->SetBranchStatus("event",1);
-   oldtree->SetBranchStatus("fNtrack",1);
-   oldtree->SetBranchStatus("fNseg",1);
-   oldtree->SetBranchStatus("fH",1);
-
-   //Create a new file + a clone of old tree in new file
-   TFile *newfile = new TFile("small.root","recreate");
-   TTree *newtree = oldtree->CloneTree();
+   const auto filename = gSystem->AccessPathName(dir) ? "./Event.root" : "$ROOTSYS/test/Event.root";
+
+   TFile oldfile(filename);
+   TTree *oldtree;
+   oldfile.GetObject("T", oldtree);
+
+   // Deactivate all branches
+   oldtree->SetBranchStatus("*", 0);
+
+   // Activate only four of them
+   for (auto activeBranchName : {"event", "fNtrack", "fNseg", "fH"})
+      oldtree->SetBranchStatus(activeBranchName, 1);
+
+   // Create a new file + a clone of old tree in new file
+   TFile newfile("small.root", "recreate");
+   auto newtree = oldtree->CloneTree();
 
    newtree->Print();
-   newfile->Write();
-   delete oldfile;
-   delete newfile;
+   newtree->Write();
```

Comment:
File->Write and TTree::Write are not interchangeable .

Reply:
But given that only a TTree is attached to the file and that the destructor of TFile invokes Close, isn't this semantically equivalent?

Reply:
TFile::Close does not invoke TFile::Write.  It might be that in this case it is sufficient but then if anything else is added to the TFile, the code will become (subtly) wrong.  I would keep TFile::Write.

Reply:
I think that relying on global registration mechanisms in our tutorials is a mistake.
It does not help transitioning our user base to the modern ROOT but reinforces more the status-quo. 
I am not going to argue about this given the number of things we need to deal withwith. Also because I am quite happy with the compromise: some of the tutorials desperately need modernisation and rewriting (these 3 are the tip of the iceberg) - any improvement counts.